### ISSUE TITLE
PCB install has garbled screen

### ISSUE THREAD

**Issue by wb1000000** [Homebrew/homebrew-x11#120]
Hi,

I used to have a working PCB from homebrew but somehow I messed up my system (something with the cairo libraries) and now I can not get it work anymore. It builds successfully but the screen is garbled...
I already did a full reinstall and upgraded to OS X 10.10, but still the same problem. Since I'm not to proficient at coding can somebody have a look to what is going on?
The only message I get in the terminal is this:

2015-08-20 11:46:00.910 pcb[4393:15057] *** WARNING: Method userSpaceScaleFactor in class NSView is deprecated on 10.7 and later. It should not be used in new applications. Use convertRectToBacking: instead. 

But I also get this message for gschem which appears to work fine.

<img width="827" alt="screen shot 2015-08-20 at 11 46 18" src="https://cloud.githubusercontent.com/assets/12801696/9380538/17a5f052-4731-11e5-91b3-c65a6e7b4270.png">

Any help greatly appreciated!

Best regards,

Jasper

**Comment by DomT4** [comment]
I can't get it to load at all here. Just hangs on initial run. What should I expect to be seeing?

**Comment by wb1000000** [comment]
Hi Dom,

I've used Macports now to install the X11 version. Here is a screenshot.
![screen shot 2015-08-26 at 21 39 35](https://cloud.githubusercontent.com/assets/12801696/9504135/30752e9a-4c3b-11e5-8de1-fe6ad6d2cc0a.png)
Did not do any further attempts using Homebrew, as it is all a bit out of my league.
Let me know if I can try anything to help.

Best regards,

Jasper

**Comment by jbattat** [comment]
I'm seeing the same issue -- any resolution?

$ 2015-08-27 16:23:42.569 pcb[50549:507] *** WARNING: Method userSpaceScaleFactor in class NSView is deprecated on 10.7 and later. It should not be used in new applications. Use convertRectToBacking: instead. 
Dynamic session lookup supported but failed: launchd did not provide a socket path, verify that org.freedesktop.dbus-session.plist is loaded!
pcb_dbus: DBus connection Error (Not enough memory)
<img width="1698" alt="screen shot 2015-08-27 at 4 23 46 pm" src="https://cloud.githubusercontent.com/assets/8468957/9532187/0b77e388-4cd8-11e5-9eee-a6d19763e25e.png">

**Comment by DomT4** [comment]
I didn't manage to get it to launch for me. I tried checking upstream but it's pretty quiet in terms of suggestion/support. I'll take another look at this over the weekend. Sorry about the trouble here :cry:

**Comment by jbattat** [comment]
ok thanks.  it may be helpful to know that the old (working) install that I had was the same version of pcb (20140316), but when it launched, only an XQuartz icon appeared on the task bar.  In the new version (20140316_1), a generic "pcb" icon appears on the task bar (so presumably it's built without XQuartz support?).
<img width="93" alt="screen shot 2015-08-29 at 7 34 14 am" src="https://cloud.githubusercontent.com/assets/8468957/9561739/677981d4-4e20-11e5-9bfe-c5255664a533.png">

so perhaps this issue is related to build-time options (does the new version use GTK/Quartz instead of XQuartz)?

**Comment by DomT4** [comment]
PCB's compile is quite unhelpful in debugging this, but my hunch is that it expects GTK+ to be built against XQuartz and have that backend available.

For the last couple of months we've no longer supported the XQuartz link in GTK+ as it complicates the whole dependency tree and makes anyone who wanted to touch GTK stuff install X11 first. I think pcb has probably quietly broken on that change. Looking into a few options.

**Comment by jbattat** [comment]
Were you able to find any path forward here? 
I downloaded and built pcb from source, but ended up with the same result (garbled screen).

For reference, here’s what I did to build pcb from source.

    Download stable snapshot (pcb-20140316)
    $ wget http://sourceforge.net/projects/pcb/files/pcb/pcb-20140316/pcb-20140316.tar.gz/download
    $ tar xvzf pcb-20140316.tar.gz

Run configure, but point to the OpenGL framework
(and I had to point to X11 libraries as well, which I know that I should not have to do, but I couldn’t get it to build otherwise)

    ./configure LDFLAGS="-framework OpenGL -L/usr/X11/lib"
    make

But make fails with errors like:

    hid/common/hidgl.c:647:49: error: use of undeclared identifier '_GLUfuncptr'
    gluTessCallback(info.tobj, GLU_TESS_ERROR,   (_GLUfuncptr)myError);

which I can work around by adding this line to src/global.h

    typedef void ( * _GLUfuncptr)(); // added by James Battat 

And then make succeeds, but with the garbled screen problem...

**Comment by 4ms** [comment]
I had the same garbled screen on OSX when installing pcb with brew. I got the same thing when compiling from source (pcb-20140316).
Finally I got it to compile without a garbled screen by disabling open-gl:

    ./configure --disable-gl
    make
    make install

<img width="1239" alt="screen shot 2015-10-14 at 9 11 49 am" src="https://cloud.githubusercontent.com/assets/1482615/10489956/adc0f418-7253-11e5-9feb-65438284e0c7.png">

**Comment by jbattat** [comment]
@4ms Thanks for suggesting a path forward.  On my machine, though, configure dies with:

```configure: error: GNU gettext tools not found; required for intltool```

So then I tried running `brew sh` first, and get a missing Perl module error:
```
$ brew sh
$ ./configure --disable-gl
...
$ checking for XML::Parser... configure: error: XML::Parser perl module is required for intltool
```

Any recommendations on how to proceed? (related:  what is the recommended perl installation?  brew?  perlbrew? CPAN?)

**Comment by oskay** [comment]
Any progress on this?

**Comment by jbattat** [comment]
Yes, sort of.
I was able to install `pcb` and `geda-gaf` from source (not through homebrew).
I was having a problem with `gettext` not being found when running `configure` for `pcb`.  For some reason, even though I installed `gettext` via `brew`, it's not on my path (see above).
```
$ brew install gettext
Warning: gettext-0.19.6 already installed
$ which brew
```
So I have to manually add `gettext` to my path:
```
$ export PATH=/usr/local/opt/gettext/bin:$PATH
```

Then, to build/install `pcb` (I'm using the `pcb-20140316` snapshot), I do:
```
$ ./configure --disable-gl --prefix /opt/geda/pcb
$ make
$ make
$ sudo make install
```
I needed to run make twice because the first make fails with an error (something like "illegal byte sequence"), but the second run works fine (???).

For `geda-gaf`, I do (using the `geda-gaf-1.8.2` release):
```
$ ./configure —prefix /opt/geda/geda-gaf
$ make
$ sudo make install
```

All that said, I still can't install either directly with `brew`.
